src/os/bluestore/BlueFS.cc

```diff
@@ -144,7 +144,7 @@ void BlueFS::add_block_extent(unsigned id, uint64_t offset, uint64_t length)
   block_all[id].insert(offset, length);
   block_total[id] += length;
 
-  if (alloc.size()) {
+  if (id < alloc.size() && alloc[id]) {
```

Comment:
Seems sage does this intentionally, we could add a block device and then  simply call `add_block_extent()` to push its space extents into **block_all**[] map, which will be properly add to the **alloc**[] map through the `_init_alloc()` method during **mkfs** or **mount** process. 

Check `test_bluefs.cc ` for detail:

```c++
TEST(BlueFS, mkfs) {
  uint64_t size = 1048476 * 128;
  string fn = get_temp_bdev(size);
  uuid_d fsid;
  BlueFS fs;
  fs.add_block_device(BlueFS::BDEV_DB, fn);
  fs.add_block_extent(BlueFS::BDEV_DB, 1048576, size - 1048576);
  fs.mkfs(fsid);
  rm_temp_bdev(fn);
}

```

So we should not assert instead.